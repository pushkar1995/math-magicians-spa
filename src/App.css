.app {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.calculator-container {
  border: 1px solid black;
  width: 30vw;
  height: 40vh;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
}

.screen {
  background-color: grey;
  color: rgb(225, 225, 225);
  padding: 0.7rem;
  height: 2.5rem;
  width: 100%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-size: 1.2rem;
  box-sizing: border-box;
}

.buttonContainer {
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

button {
  border: 1px solid grey;
  background-color: rgb(212, 212, 212);
  color: rgb(39, 60, 60);
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
}

.button-container > *:nth-child(4n+4),
.equal {
  background-color: orange;
}
