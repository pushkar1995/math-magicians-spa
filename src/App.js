import './App.css';
import Calculator from './components/calculator.js';

function App() {
  return (
    <div className="app">
      <Calculator />
    </div>
  );
}

export default App;