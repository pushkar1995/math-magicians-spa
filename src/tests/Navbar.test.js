import React from 'react';

import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

import NavBar from '../components/Navbar/Navbar';

import App from '../App';

describe('Test NavigationBar', () => {
  test('Render correctly', () => {
    const tree = render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });

  describe('Click events', () => {
    // test('User clicks Quotes', () => {
    //   const { container } = render(<App />);
    //   expect(screen.getByText('Welcome to our page!')).toBeTruthy();
    //   expect(container.getElementsByClassName('quoteDiv').length).toBe(0);

    //   const quoteLinks = screen.getAllByText('Quotes');
    //   userEvent.click(quoteLinks[0]); // Assuming the first one is the desired link
    //   expect(container.getElementsByClassName('quoteDiv').length).toBe(1);
    // });

    // test('User leaves Home and then clicks Home again', () => {
    //   render(<App />);
    //   userEvent.click(screen.getAllByText('Calculator')[0]);
    //   expect(screen.queryByText('Welcome to our page!')).not.toBeTruthy();
    //   userEvent.click(screen.getByText('Home'));
    //   expect(screen.getByText('Welcome to our page!')).toBeTruthy();
    // });
    // test('User clicks Calculator', () => {
    //   const { container } = render(<App />);
    //   expect(screen.queryByText("Let's do some math!")).toBeNull();
    //   userEvent.click(screen.getAllByText('Calculator')[0]);
    //   expect(screen.getByText("Let's do some math!")).toBeTruthy();
    //   expect(container.getElementsByClassName('calc').length).toBe(1);
    // });
  });
});
