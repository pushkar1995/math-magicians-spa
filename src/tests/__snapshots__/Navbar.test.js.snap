// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NavigationBar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h1>
          Math Magicians
        </h1>
        <ul>
          <li>
            <a
              class="links"
              href="/"
            >
               
              Home
               
            </a>
          </li>
          <li>
            <a
              class="links"
              href="/calculator"
            >
               
              Calculator
               
            </a>
          </li>
          <li>
            <a
              class="links"
              href="/quotes"
            >
               
              Quotes
               
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h1>
        Math Magicians
      </h1>
      <ul>
        <li>
          <a
            class="links"
            href="/"
          >
             
            Home
             
          </a>
        </li>
        <li>
          <a
            class="links"
            href="/calculator"
          >
             
            Calculator
             
          </a>
        </li>
        <li>
          <a
            class="links"
            href="/quotes"
          >
             
            Quotes
             
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
